### PR TITLE
Update flink-demo to use operator dependencies

### PR DESCRIPTION
**Pull request by zen-dog**
Summary:
- using `KudoOperator` task to include Zookeeper, Kafka and Flink dependencies directly instead of pre-installing them manually (and activate using `Instance` resources)
- the corresponding parameters are in the params files e.g. `templates/kafka-params.yaml`
- README is updated to reflect the latest changes
- demo relies on not yet releases kudoctl and manager

Signed-off-by: Aleksey Dukhovniy <alex.dukhovniy@googlemail.com>

### CODE REVIEW

File: repository/flink/docs/demo/financial-fraud/README.md [hunk 1]
```diff
@@ -42,50 +42,25 @@ $ minikube start --vm-driver=hyperkit --cpus=6 --memory=9216 --disk-size=10g
 
 ### Install KUDO and the KUDO CLI
 
-- Have current KUDO CLI `v0.10.x` [installed](https://kudo.dev/docs/cli.html#setup-the-kudo-kubectl-plugin)
+- Have current KUDO CLI `v0.15.x` [installed](https://kudo.dev/docs/cli.html#setup-the-kudo-kubectl-plugin)
     - Run `kubectl kudo version`, the output should look like:
         ```bash
         $ kubectl kudo version
         KUDO Version: version.Info{GitVersion:"0.10.0", GitCommit:"d2310b12", BuildDate:"2020-01-10T18:54:59Z", GoVersion:"go1.13.6", Compiler:"gc", Platform:"darwin/amd64"}
```
Comment (kensipe) [inline repository/flink/docs/demo/financial-fraud/README.md]: the output example still shows `0.10.0`.  It makes sense that we don't have the build date or sha as we haven't had a release for this version... but it doesn't make sense to change the version from `v0.10.x` to `v0.15.x` and not change the gitVersion to `"0.15.0"`

**Review by kensipe — CHANGES_REQUESTED** [review]
My tests needed to deviate because things are not released... details below (but I believe this should work)... the end result issue:

```
kubectl kudo plan status --instance flink-demo
Plan(s) for "flink-demo" in namespace "default":
.
└── flink-demo (Operator-Version: "flink-demo-0.1.5" Active-Plan: "deploy")
    └── Plan deploy (serial strategy) [FATAL_ERROR], last updated 2020-07-01 08:37:22
        ├── Phase dependencies (serial strategy) [FATAL_ERROR]
        │   ├── Step zookeeper [FATAL_ERROR] (default/flink-demo fatal error:  failed to build task deploy.dependencies.zookeeper.zookeeper: task validation error: kudo operator task 'zookeeper' has an empty operator name)

```

Deviations:
1. running local manager with `KUDO_CERT_DIR=./test/cert/ make run`
2. Installing from the local files system:  `go run cmd/kubectl-kudo/main.go install ../operators/repository/flink/docs/demo/financial-fraud/demo-operator/ --instance flink-demo`

File: repository/flink/docs/demo/financial-fraud/README.md (hunk 1 above)
Comment (zen-dog) [reply repository/flink/docs/demo/financial-fraud/README.md]: Good catch. I missed it

**Review by zmalik — APPROVED** [review]
LGTM!

**Comment by zmalik** [comment]
and tested it with `0.15.0-rc1`
```
kubectl kudo plan status --instance=flink-demo
Plan(s) for "flink-demo" in namespace "default":
.
└── flink-demo (Operator-Version: "flink-demo-0.1.5" Active-Plan: "deploy")
    └── Plan deploy (serial strategy) [COMPLETE], last updated 2020-07-09 11:32:20
        ├── Phase dependencies (serial strategy) [COMPLETE]
        │   ├── Step zookeeper [COMPLETE]
        │   └── Step kafka [COMPLETE]
        ├── Phase flink-cluster (serial strategy) [COMPLETE]
        │   └── Step flink [COMPLETE]
        ├── Phase demo (serial strategy) [COMPLETE]
        │   ├── Step gen [COMPLETE]
        │   └── Step act [COMPLETE]
        └── Phase flink-job (serial strategy) [COMPLETE]
            └── Step submit [COMPLETE]
```

**Review by kensipe — APPROVED** [review]
looks good!

**Pull request merged**